server/src/main/java/io/spine/server/entity/DefaultLifecycle.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.entity;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+import io.spine.client.EntityId;
+import io.spine.core.Command;
+import io.spine.core.CommandId;
+import io.spine.core.Event;
+import io.spine.core.EventId;
+import io.spine.option.EntityOption;
+import io.spine.system.server.ArchiveEntity;
+import io.spine.system.server.AssignTargetToCommand;
+import io.spine.system.server.ChangeEntityState;
+import io.spine.system.server.CommandTarget;
+import io.spine.system.server.CreateEntity;
+import io.spine.system.server.DeleteEntity;
+import io.spine.system.server.DispatchCommandToHandler;
+import io.spine.system.server.DispatchEventToReactor;
+import io.spine.system.server.DispatchEventToSubscriber;
+import io.spine.system.server.DispatchedMessageId;
+import io.spine.system.server.EntityHistoryId;
+import io.spine.system.server.ExtractEntityFromArchive;
+import io.spine.system.server.MarkCommandAsHandled;
+import io.spine.system.server.RestoreEntity;
+import io.spine.system.server.SystemGateway;
+import io.spine.type.TypeUrl;
+
+import java.util.Collection;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.base.Identifier.pack;
+import static io.spine.util.Exceptions.newIllegalArgumentException;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * The default implementation of {@link Repository.Lifecycle}.
```

Comment:
Please say something on what "default" means and what the scenarios are, for which the default is not an option (hence, a "custom" is needed).

Reply:
Done.